### PR TITLE
magento/magento2#28957: GraphQl. Mutation: customerOrders query and gift_message attribute

### LINKED ISSUE #28957: GraphQl. Mutation: customerOrders query and gift_message attribute
### Preconditions (*)

Error `Can't load gift message for order` is returned when order does not have a `gift_message`:

### Steps to reproduce (*)

1. Magento (2.4-develop) with a sample data
1. Generate customer token:

```
mutation {
  generateCustomerToken(
    email: "roni_cost@example.com"
    password: "roni_cost3@example.com"
  ) {
    token
  }
}
```

1. Retrieve customer orders:

```
query {
    customerOrders {
        items {
            order_number
            gift_message {
                to
                from
                message
            }
        }
    }
}
```

### Expected result (*)

We do not need to have a `Can't load gift message for order` error in the response when an order does not have `gift_message`.

### Actual result (*)

There is a `Can't load gift message for order` error in the response when an order does not have `gift_message`.

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by atwixfirster**
### Description (*)

PR fixes https://github.com/magento/magento2/issues/28957

### Fixed Issues (if relevant)

1. Fixes magento/magento228957: GraphQl. Mutation: customerOrders query and gift_message attribute

### Manual testing scenarios (*)
1. Magento (2.4-develop) with a sample data
1. Generate customer token:

```
mutation {
  generateCustomerToken(
    email: "roni_cost@example.com"
    password: "roni_cost3@example.com"
  ) {
    token
  }
}
```

1. Retrieve customer orders:

```
query {
    customerOrders {
        items {
            order_number
            gift_message {
                to
                from
                message
            }
        }
    }
}
```

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by atwixfirster** [comment]
@magento run all tests

**Comment by atwixfirster** [comment]
@magento run Database Compare

**Comment by atwixfirster** [comment]
@magento run WebAPI Tests

**Comment by atwixfirster** [comment]
@magento run Unit Tests

**Comment by atwixfirster** [comment]
@magento Static Tests

**Comment by atwixfirster** [comment]
@magento run Semantic Version Checker

**Comment by atwixfirster** [comment]
@magento run Functional Tests B2B

**Comment by atwixfirster** [comment]
@magento run Sample Data Tests EE  Sample Data Tests CE Sample Data Tests B2B Magento Health Index

**Review by dmytro-ch — CHANGES_REQUESTED** [review]
Hello @atwixfirster, thank you for your contribution.
The overall changes look good to me.
Could you please cover this case by an appropriate functional test?

**Comment by sivaschenko** [comment]
Hi @nrkapoor , what is the reason for putting this PR on hold? Should it be processed according to priority or any additional actions are required?

**Comment by sivaschenko** [comment]
`customerOrders` query is `deprecated`: https://github.com/magento/magento2/blob/8b1d1184e67f31268a38cb8ec281e6c48b33d43d/app/code/Magento/SalesGraphQl/etc/schema.graphqls#L5

Need to check if the issue is reproducible with `customer orders` and update the issue and PR testing steps accordingly

File: app/code/Magento/GiftMessageGraphQl/Model/Resolver/Order/GiftMessage.php [hunk 1]
```diff
@@ -55,17 +55,13 @@ public function resolve(
...
-
         try {
             $orderGiftMessage = $this->orderRepository->get($value['id']);
         } catch (LocalizedException $e) {
-            throw new GraphQlInputException(__('Can\'t load gift message for order'));
+            return null;
         }
 
-        if (!isset($orderGiftMessage)) {
+        if (!$orderGiftMessage->getGiftMessageId()) {
```
Comment (cpartica) [inline app/code/Magento/GiftMessageGraphQl/Model/Resolver/Order/GiftMessage.php:64]: I think this change needs a test, plus we need to fix webAPI tests which are failing

**Review by cpartica — CHANGES_REQUESTED** [review]

**Comment by engcom-Alfa** [comment]
We raised internal PR- https://github.com/magento-gl/magento2ce/pull/290 along with this PR commits, covering the automation tests and resolved PR comments.
Once Internal PR will get merged will notify here

**Pull request merged**